velox/exec/tests/TreeOfLosersTest.cpp

```diff
@@ -100,3 +100,144 @@ TEST_F(TreeOfLosersTest, singleWithEquals) {
     result.first->pop();
   }
 }
+
+TEST_F(TreeOfLosersTest, allDuplicates) {
+  const int kNumsPerStream = 40;
+  const int kNumStreams = 20;
+  const uint32_t kValue = 10;
+  for (bool testNextEqual : {false, true}) {
+    SCOPED_TRACE(fmt::format("testNextEqual: {}", testNextEqual));
+    std::vector<std::unique_ptr<TestingStream>> mergeStreams;
+    for (int i = 0; i < kNumStreams; ++i) {
+      std::vector<uint32_t> streamNumbers;
+      for (int j = 0; j < kNumsPerStream; ++j) {
+        streamNumbers.push_back(kValue);
+      }
+      mergeStreams.push_back(
+          std::make_unique<TestingStream>(std::move(streamNumbers)));
+    }
+    TreeOfLosers<TestingStream> merge(std::move(mergeStreams));
+    for (auto i = 0; i < kNumStreams * kNumsPerStream; ++i) {
+      TestingStream* stream;
+      if (testNextEqual) {
+        auto result = merge.nextWithEquals();
+        stream = result.first;
+        // NOTE: the last stream has no other stream with equal value.
+        if (i < (kNumStreams - 1) * kNumsPerStream) {
+          ASSERT_TRUE(result.second) << i;
+        } else {
+          ASSERT_FALSE(result.second) << i;
+        }
+      } else {
+        stream = merge.next();
+      }
+      ASSERT_TRUE(stream != nullptr) << i;
+      ASSERT_EQ(stream->current()->value(), kValue) << i;
+      stream->pop();
+    }
+  }
+}
+
+TEST_F(TreeOfLosersTest, allSorted) {
+  const int kNumsPerStream = 40;
+  const int kNumStreams = 20;
+  const uint32_t kStartValue = 10;
+  for (bool testNextEqual : {/*false, */ true}) {
```

Comment:
We can test `next()` as well here

Reply:
Yeah, I commented out while debugging the test code. Thanks for pointing out!